Applications/Spire/Include/Spire/Ui/DropDownItem.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef SPIRE_DROP_DOWN_ITEM_HPP
+#define SPIRE_DROP_DOWN_ITEM_HPP
+#include <QImage>
+#include <QWidget>
+#include "Spire/Spire/Spire.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+
+namespace Spire {
+
+  //! Represents an item used in a DropDownList.
+  class DropDownItem : public QWidget {
+    public:
+
+      //! Signals that the item was highlighted with the mouse.
+      /*!
+        \param value The value of the DropDownItem.
+      */
+      using HighlightedSignal = Signal<void (QVariant value)>;
```

Comment:
Can you explain again why a ```const &``` isn't being used for these signals?

Reply:
The issue was that the item might be removed from the list in response to the signal, so anything else listening to the signal may receive a reference to an object that was destroyed.

Reply:
That's fine, what you do in this case is something along the lines of:

    auto value = somevalue;
    m_highlighted_signal(value);

In other words, whatever calls the signal makes a single copy and then passes a reference to that copy to the signal. This avoids every receiver from having to make a copy.